### ISSUE TITLE
TimeZoneId can cause difficult-to-track error if JVM used is incompatible

### ISSUE THREAD

**Issue by renatoathaydes** [iZettle/izettle-toolbox#250]
The [TimeZoneId](https://github.com/iZettle/izettle-toolbox/blob/master/izettle-java/src/main/java/com/izettle/java/TimeZoneId.java) class currently enumerates all time-zones in the world.

It counts on being up-to-date with the JVM's own timezone registry, whose change log can be seen [here](http://www.oracle.com/technetwork/java/javase/tzdata-versions-138805.html). This registry changes frequently, it seems.

When we deploy a new version of this class and do not also remember to modify the JVM version we use in the Docker image, a class initialization error can occur.

This has happened in different projects, it seems. Lastly, in the authorization server release, which was running on an older JVM (8 update 112) and upgraded the version it was using of this class (via updating the `izettle-toolbox` version it depended upon).

The error was hard to track because it happened in an asynchronous event publisher (as it tried to serialize the message to be published) and did not directly show up in the Splunk logs.

For posterity, here's what the error looks like:

```
Caused by: java.time.zone.ZoneRulesException: Unknown time-zone ID: Asia/Famagusta
	at java.time.zone.ZoneRulesProvider.getProvider(ZoneRulesProvider.java:272)
	at java.time.zone.ZoneRulesProvider.getRules(ZoneRulesProvider.java:227)
	at java.time.ZoneRegion.ofId(ZoneRegion.java:120)
	at java.time.ZoneId.of(ZoneId.java:411)
	at java.time.ZoneId.of(ZoneId.java:359)
	at com.izettle.java.TimeZoneId.toZoneId(TimeZoneId.java:648)
	at com.izettle.java.TimeZoneId.lambda$static$0(TimeZoneId.java:623)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at com.izettle.java.TimeZoneId.<clinit>(TimeZoneId.java:621)
```

I got this by intentionally reproducing the error on my machine using the old JVM.

Not straightforward to see the connection.

My suggestion is to use the JVM values without enumerating the values ourselves. If we need certain well-known values to be enumerated, maybe we should enumerate only those, and fail only if the well known timezones cannot be located.

**Comment by adamvoncorswant** [comment]
edit: reduced the stack trace to the relevant parts.

**Comment by renatoathaydes** [comment]
Fixed in bfda4bdd0771485137f7d59d3ba87c92135ee52c.
